pymc_experimental/tests/test_laplace.py

```diff
@@ -0,0 +1,65 @@
+#   Copyright 2024 The PyMC Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+
+import numpy as np
+import pymc as pm
+import pytest
+
+import pymc_experimental as pmx
+
+
+@pytest.mark.filterwarnings(
+    "ignore:Model.model property is deprecated. Just use Model.:FutureWarning",
+    "ignore:hessian will stop negating the output in a future version of PyMC.\n"
+    + "To suppress this warning set `negate_output=False`:FutureWarning",
+)
+def test_laplace():
+
+    # Example originates from Bayesian Data Analyses, 3rd Edition
+    # By Andrew Gelman, John Carlin, Hal Stern, David Dunson,
+    # Aki Vehtari, and Donald Rubin.
+    # See section. 4.1
+
+    y = np.array([2642, 3503, 4358], dtype=np.float64)
+    n = y.size
+    draws = 100000
+    chains = 1
+
+    with pm.Model() as m:
+        logsigma = pm.Uniform("logsigma", 1, 100)
+        mu = pm.Uniform("mu", -10000, 10000)
+        yobs = pm.Normal("y", mu=mu, sigma=pm.math.exp(logsigma), observed=y)
+        vars = [mu, logsigma]
```

Comment:
Question: say you only did `vars=[mu]`, how would the variable `logsigma` be estimated?

Reply:
I think find_MAP in that case uses the initial_point for the excluded variable. I never found that behavior useful tbh

Reply:
Sounds like we may want to pass a similar `start` kwarg to laplace to set the value of variables that are not being optimized?

Reply:
worth adding a test on this to confirm the behaviour

Reply:
I am not sure I fully understand this. I will give it a second go with the documentation for find_MAP.

Reply:
Yeah, just a little confirmation of what happens in the test case when you don't pass all the variables into `pm.fit` e.g. `vars = [logsigma]`. Doesn't need to be a full test if it doesn't break things, it can be a warning or a line in the docs at this point.

Rest of the PR looks great. Nice job

Reply:
@theorashid it sound like a good idea with a warning in case one does not pass all variable into the fit method. I was going through the documentation and wondering if the PYMC model attribute 'value_vars' is the proper attribute to use to check if number of variables in 'vars' is the same as all the variables in the model.

Reply:
Hi @theorashid. I am not sure how to handle if only a subset of the variables are passed in, e.g. vars=[mu] and log_sigma is left out. If this should raise a warning I need some way of figuring out the number of model parameters and compare that with the number of parameters in vars. I am not sure how to determine the number of model parameters

Reply:
`model.free_RVs`

Reply:
@theorashid and @ricardoV94, I committed an update that will raise a warning in case number of variables in vars does not equal number of model variables.

Reply:
LGTM